Subject: [PATCH] Camera service: Speed up camera service startup on boot

In the common case, cameraserver starts before system server, which
contains the proxy camera service.  This means that
getCameraServiceProxy() will wait for the proxy to be up (or time out
after 5 seconds).  Since the proxy will call the camera service once it
is up and running, there's no reason to block.

Switch to checkService() instead of getService() to remove this timeout
and fail immediately.
Test: Manual: Build and boot phone; confirm that camera functions normally

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index 85faac6..86c48a4 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -273,7 +273,10 @@
 
 sp<ICameraServiceProxy> CameraService::getCameraServiceProxy() {
     sp<IServiceManager> sm = defaultServiceManager();
-    sp<IBinder> binder = sm->getService(String16("media.camera.proxy"));
+    // Use checkService because cameraserver normally starts before the
+    // system server and the proxy service. So the long timeout that getService
+    // has before giving up is inappropriate.
+    sp<IBinder> binder = sm->checkService(String16("media.camera.proxy"));
     if (binder == nullptr) {
         return nullptr;
     }
